File: src/jk_5/nailed/map/TeamManager.java
package jk_5.nailed.map;

import com.google.common.collect.Sets;
import jk_5.nailed.config.helper.ConfigFile;
import jk_5.nailed.config.helper.ConfigTag;
import jk_5.nailed.teams.Team;
import jk_5.nailed.util.EnumColor;
import net.minecraft.src.ScorePlayerTeam;
import net.minecraft.src.Scoreboard;

import java.util.List;
import java.util.Set;

/**
 * No description given
 *
 * @author jk-5
 */
public class TeamManager {

    private final Map map;
    private Set<Team> teams = Sets.newHashSet();

    public ScorePlayerTeam spectatorTeam;

    public TeamManager(Map map) {
        this.map = map;
    }

    public void setupTeams() {
        ConfigFile conf = this.map.getMappack().getConfig();
        List<ConfigTag> teams = conf.getTag("teams").getSortedTagList();
        if (teams == null) return;
        Scoreboard s = this.map.getWorld().getScoreboard();
        for (ConfigTag tag : teams) {
            EnumColor color = EnumColor.valueOf(tag.getTag("color").getValue("white").toUpperCase());
            String name = tag.getTag("name").getValue("");
            Team team = new Team(name, tag.name(), color);
            team.setMap(this.map);
            this.teams.add(team);
            ScorePlayerTeam scoreboardTeam = s.func_96527_f(tag.name());                            //teamid
            scoreboardTeam.func_96664_a(name);                                                      //teamname
            scoreboardTeam.func_96660_a(tag.getTag("frienlyfire").getBooleanValue(false));          //friendlyfire
            scoreboardTeam.func_98300_b(tag.getTag("friendlyinvisibles").getBooleanValue(true));    //friendlyinvisibles
            scoreboardTeam.func_96666_b(color.toString());                                          //teamprefix
            scoreboardTeam.func_96662_c(EnumColor.RESET.toString());                                //teamsuffix
            team.scoreboardTeam = scoreboardTeam;
        }
        this.createSpectatorTeam();
    }

    private void createSpectatorTeam() {
        Scoreboard s = this.map.getWorld().getScoreboard();
        spectatorTeam = s.func_96527_f("spectator");
        spectatorTeam.func_96664_a("Spectator");                   //teamname
        spectatorTeam.func_96660_a(false);                         //friendlyfire
        spectatorTeam.func_98300_b(true);                          //friendlyinvisibles
        spectatorTeam.func_96666_b(EnumColor.AQUA.toString());     //teamprefix
        spectatorTeam.func_96662_c(EnumColor.RESET.toString());    //teamsuffix
    }

    public Team getTeam(String teamId) {
        for (Team team : this.teams) {
            if (team.getTeamID().equals(teamId)) {
                return team;
            }
        }
        return null;
    }

    public Set<String> getTeamNames() {
        Set<String> names = Sets.newHashSet();
        for (Team team : this.teams) {
            names.add(team.getTeamID());
        }
        return names;
    }
}

File: src/jk_5/nailed/map/TeamManager.scala
package jk_5.nailed.map

import scala.collection.{JavaConversions, mutable}
import jk_5.nailed.team.Team
import net.minecraft.src.ScorePlayerTeam
import jk_5.nailed.util.EnumColor

/**
 * No description given
 *
 * @author jk-5
 */
class TeamManager(private final val map: Map) {

  private final val teams = new mutable.HashSet[Team]()
  private var spectatorTeam: ScorePlayerTeam = _

  def setupTeams(){
    val conf = this.map.getMappack.getConfig
    val teamlist = conf.getTag("teams").getSortedTagList
    if(teamlist == null) return
    val scoreboard = this.map.getScoreboard
    for(tag <- JavaConversions.collectionAsScalaIterable(teamlist)){
      val team = new Team(tag.name)
      val color = EnumColor.valueOf(tag.getTag("color").getValue("white").toUpperCase)
      team.setName(tag.getTag("name").getValue(""))
      team.setColor(color)
      team.setMap(this.map)
      team.setFriendlyFireEnabled(tag.getTag("friendlyfire").getBooleanValue(default = false))
      this.teams.add(team)
      val scoreboardTeam = scoreboard.createTeam(tag.name)
      scoreboardTeam.setTeamName(tag.getTag("name").getValue(""))
      scoreboardTeam.setAllowFriendlyFire(team.friendlyFireEnabled)
      scoreboardTeam.setSeeFriendlyInvisiblesEnabled(tag.getTag("frienlyinvisibles").getBooleanValue(default = true))
      scoreboardTeam.setNamePrefix(color.toString)
      scoreboardTeam.setNameSuffix(EnumColor.RESET.toString)
      team.setScoreboardTeam(scoreboardTeam)
    }
    this.createSpectatorTeam()
  }

  private def createSpectatorTeam(){
    this.spectatorTeam = this.map.getScoreboard.createTeam("spectator")
    this.spectatorTeam.setTeamName("Spectator")
    this.spectatorTeam.setAllowFriendlyFire(false)
    this.spectatorTeam.setSeeFriendlyInvisiblesEnabled(true)
    this.spectatorTeam.setNamePrefix(EnumColor.AQUA.toString)
    this.spectatorTeam.setNameSuffix(EnumColor.RESET.toString)
  }

  @inline def getTeam(teamId: String) = this.teams.find(_.getTeamID == teamId)
  @inline def getTeamNames = this.teams.map(_.getTeamID)
  @inline def getTeams = this.teams
  @inline def getSpectatorTeam = this.spectatorTeam
}
